includes/discovery/sensors/load/primekey.inc.php

```diff
@@ -0,0 +1,90 @@
+<?php
+/*
+ * LibreNMS load sensor for PrimeKey Hardware Appliances
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.  Please see LICENSE.txt at the top level of
+ * the source code distribution for details.
+ *
+ * The OIDs described here, for the EJBCA Appliance:
+ * https://doc.primekey.com/ejbca-appliance/operations/webconf-configurator-of-hardware-appliance/monitoring
+ *
+ * @package    LibreNMS
+ * @subpackage webui
+ * @link       https://www.librenms.org
+ * @copyright  2022 LibreNMS
+ * @author     LibreNMS Contributors
+ */
+
+$oids = [
+    50 => [
+        'descr' => ' 1min average',
+        'oid'   => '.1.3.6.1.4.1.22408.1.1.2.1.5.108.111.97.100.50.1',
+        'group' => 'CPU',
+    ],
+    51 => [
+        'descr' => ' 5min average',
+        'oid'   => '.1.3.6.1.4.1.22408.1.1.2.1.5.108.111.97.100.51.1',
+        'group' => 'CPU',
+    ],
+    52 => [
+        'descr' => '15min average',
+        'oid'   => '.1.3.6.1.4.1.22408.1.1.2.1.5.108.111.97.100.52.1',
+        'group' => 'CPU',
+    ],
+];
+
+$class = 'load';
+
+$type = 'primekey';
+$divisor = 1;
+$multiplier = 100;
```

Comment:
Here we're checking the system load averages as reported by "uptime" in Linux, generally a number between 0 and 1.
We multiply by 100 which should provide approximately a percentage - assuming a single threaded system.

The appliance I'm testing against apparently has an "Intel(R) Xeon(R) CPU E3-1225 v5" processor - i.e. 4 Cores, 4 threads.

On that bases I'm not sure if I should set the multiplier to 25, on the basis that a system load of 1.0 would indicate that the CPU is 25% (one quarter) full.

Reply:
Hi
System load is a value "per se" and should not be modified.

Reply:
Load is displayed by Librenms as a percentage, with certain default warning and thresholds

Reply:
Load as sensor in LibreNMS is a percentage and is used for UPSes etc.

Usually, CPU usage is reported with the "processor" module and not the "sensor" module. In the processor module, you can choose the source of the data, and the ratio to apply on it. 

So, 2 variants are: 
- Either you use the load as reported by the linux device, and display it with a "count" sensor, without changing it. 
- Or you use the processor module and use the load value with a maximum defined as "cpucount". BUT if you do so, then any load above cpucount will result in a percentage over 100% which does not make sense. 

That's why a load value should be kept as it is.

Reply:
Checked with the vendor and for now I think setting "cpucount" to 4 will work for all the appliances they sell currently. 

I've not yet found another reference to cpucount in this repo, however on [this page](https://docs.librenms.org/Developing/os/Mem-CPU-Information/#yaml_1) I see a yaml data key called precision:

> <html>
<body>
<!--StartFragment-->

precision | 1 | The multiplier to multiply the data by. If this is negative, the data will be multiplied then subtracted from 100.
-- | -- | --


<!--EndFragment-->
</body>
</html>
Having taken a quick rummage around the LibreNMS/OS folder I think I need to find and better comprehend the Processor::discover function before I can go down the processor module route.....

TL;DR: Should I pick your first option and display it as an unaltered "count" sensor?  😁